pkg/reconciler/autoscaling/hpa/hpa_test.go

```diff
@@ -486,3 +498,22 @@ func deploy(namespace, name string, opts ...deploymentOption) *appsv1.Deployment
 	}
 	return s
 }
+
+func defaultConfig() *config.Config {
+	autoscalerConfig, _ := autoscaler.NewConfigFromMap(nil)
+	return &config.Config{
+		Autoscaler: autoscalerConfig,
+	}
+}
+
+type testConfigStore struct {
```

Comment:
Perhaps just `type testConfigStore *config.Config`?

Reply:
As discussed in Slack: This gets kinda hairy as I need cannot use the pointer type as the aliased type. I'd end up with `type testConfigStore config.Config`. All of the functions will remain pointer receivers though.

In the `ToContext` method I'll need to cast the value so the compiler eats it. On the caller side I'll need to define extra variables to be able to get a pointer to this new type.

Here's what the diff would look like:

```diff
diff --git a/pkg/reconciler/autoscaling/hpa/hpa_test.go b/pkg/reconciler/autoscaling/hpa/hpa_test.go
index 0af6c524..74de5a22 100644
--- a/pkg/reconciler/autoscaling/hpa/hpa_test.go
+++ b/pkg/reconciler/autoscaling/hpa/hpa_test.go
@@ -415,13 +415,14 @@ func TestReconcile(t *testing.T) {
 	}}
 
 	defer logtesting.ClearAll()
+	configStore := testConfigStore(*defaultConfig())
 	table.Test(t, MakeFactory(func(listers *Listers, opt reconciler.Options) controller.Reconciler {
 		return &Reconciler{
 			Base:        reconciler.NewBase(opt, controllerAgentName),
 			paLister:    listers.GetPodAutoscalerLister(),
 			sksLister:   listers.GetServerlessServiceLister(),
 			hpaLister:   listers.GetHorizontalPodAutoscalerLister(),
-			configStore: &testConfigStore{config: defaultConfig()},
+			configStore: &configStore,
 		}
 	}))
 }
@@ -506,12 +507,10 @@ func defaultConfig() *config.Config {
 	}
 }
 
-type testConfigStore struct {
-	config *config.Config
-}
+type testConfigStore config.Config
 
 func (t *testConfigStore) ToContext(ctx context.Context) context.Context {
-	return config.ToContext(ctx, t.config)
+	return config.ToContext(ctx, (*config.Config)(t))
 }
 
 func (t *testConfigStore) WatchConfigs(w configmap.Watcher) {}
```